### PR TITLE
feat(csrc): Add residual indices support with uint8_t and validate quant-GEMV.

### PR DESCRIPTION
**Pull request by lcy-seso**
* Enhanced the data loading to ensure all input data are processed correctly.
* Added residual indices support with uint8_t.
* Added a PyTorch-based unit test to validate the computational results, ensuring the accuracy of Quant-GEMV operations.

### CODE REVIEW

**Comment by wejoncy** [comment]
Cool, Do we have some benchmark numbers?

**Comment by lcy-seso** [comment]
> Cool, Do we have some benchmark numbers?

I'm working on this and will have numbers shortly.

In this implementation, all codebooks are stored in shared memory—with precautions to avoid bank conflicts—while all intermediate results are kept in each thread's local register file.

Performance is influenced by several tunable hyperparameters, and I'm currently evaluating their effects.
